contributing/development/cpp_usage_guidelines.rst

```diff
@@ -86,6 +86,8 @@ scripting API.
 | ``List``               | ``std::list``            | Linked list type. Generally slower than other array/vector types. Prefer using        |
 |                        |                          | other types in new code, unless using ``List`` avoids the need for type conversions.  |
 +------------------------+--------------------------+---------------------------------------------------------------------------------------+
+| ``FixedVector``        | ``std::array``           | Vector with a fixed capacity (more similar to ``boost::container::static_vector``).   |
```

Comment:
I think we should mention that the data is allocated on the stack so there are no heap allocations

Reply:
I didn't find a good formatulation for it, but "makes no heap allocations" is great! I'll add that.